### PR TITLE
Add kernel32 functions

### PR DESCRIPTION
**Pull request by skdltmxn**
Added most of kernel32 functions and related structures.

### CODE REVIEW

**Comment by skdltmxn** [comment]
it seems there are some unsolved errors, i will reopen PR as soon as i resolve them all

File: src/winbase.rs [hunk 1]
```diff
@@ -162,3 +162,421 @@ pub const THREAD_PRIORITY_IDLE: ::DWORD = THREAD_BASE_PRIORITY_IDLE;
...
+}
+BITFIELD!(COMSTAT BitFields: ::DWORD [
+    fCtsHold set_fCtsHold[0..1],
+    fDsrHold set_fDsrHold[1..2],
+    fRlsdHold set_fRlsdHold[2..3],
+    fXoffHold set_fXoffHold[3..4],
+    fXoffSent set_fXoffSent[4..5],
+    fEof set_fEof[5..6],
+    fTxim set_fTxim[6..7],
+    fReserved set_fReserved[7..25],
```
Comment (retep998) [inline src/winbase.rs]: `7..32`

File: src/winbase.rs [hunk 2]
```diff
@@ -162,3 +162,421 @@ pub const THREAD_PRIORITY_IDLE: ::DWORD = THREAD_BASE_PRIORITY_IDLE;
...
+pub type LPCOMMCONFIG = *mut COMMCONFIG;
+
+pub type PFIBER_CALLOUT_ROUTINE = Option<unsafe extern "system" fn(
+    lpParameter: ::LPVOID,
+) -> ::LPVOID>;
+pub type LPLDT_ENTRY = ::LPVOID;
+
+#[repr(i32)] #[derive(Clone, Copy, Debug)]
+pub enum COPYFILE2_MESSAGE_TYPE {
+	COPYFILE2_CALLBACK_NONE = 0,
```
Comment (retep998) [inline src/winbase.rs]: Incorrect indentation.

**Comment by skdltmxn** [comment]
I recreated PR since I did a force push. (mistakes fixed)
Is there any way to delete this PR?

**Comment by retep998** [comment]
As far as I know there is no way to delete issues/PRs.
Also, there's no need to create a new PR unless you've deleted your repo and forked it again, at which point the original PR is no longer tied to your repo which makes updating it impossible.

**Comment by skdltmxn** [comment]
Well, I recreated PR because github said I cannot reopen it since there was a force push.

**Comment by retep998** [comment]
Well, I should have still been able to reopen it myself regardless, but now I can't because you have another PR from the same branch. Next time, just don't close it but instead leave a message saying not to merge yet.

**Comment by skdltmxn** [comment]
Alright, I got that.

Git is hard...

**Comment by retep998** [comment]
Whenever you're neck-deep in lifetimes, trying to get some Rust code to work, just remember that Git could be even worse :stuck_out_tongue:

**Comment by skdltmxn** [comment]
Now, that's a lot comforting :grinning:

**Pull request closed without merging**